Gems/Atom/Tools/MaterialCanvas/Assets/MaterialCanvas/GraphData/Nodes/Textures/sample_texture_2d.materialgraphnode

```diff
@@ -13,7 +13,7 @@
                 "Atom/Features/ColorManagement/TransformColor.azsli"
             ],
             "instructions": [
-                "SLOTTYPE(outColor) outColor = TransformColor4(MaterialSrg::inValue.Sample(MaterialSrg::inSampler, (float2)inUV), inColorSpaceFrom, inColorSpaceTo);"
+                "SLOTTYPE(outColor) outColor = TransformColor4(asTexture(params.inValue).Sample(GetMaterialTextureSampler(), (float2)inUV), inColorSpaceFrom, inColorSpaceTo);"
```

Comment:
This is a breaking change. Material Canvas supports creating and customizing texture samplers, which get injected into the material SRG. If that is no longer generated then proper support needs to be added somewhere else for creating and customizing texture samplers. Users have existing content with different wrap and filter modes and other settings that they cannot create with the handful of PBR materials we provide.

Reply:
added Texture Samplers State to MaterialProperties, and register them in a `m_samplers` - array in the srg